tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -892,32 +892,23 @@ SpirvInstruction *SpirvEmitter::loadIfGLValue(const Expr *expr,
         const auto fromType =
             vecSize == 1 ? uintType
                          : astContext.getExtVectorType(uintType, vecSize);
-        const auto toType =
-            vecSize == 1 ? boolType
-                         : astContext.getExtVectorType(boolType, vecSize);
         loadedInstr =
-            castToBool(loadedInstr, fromType, toType, expr->getLocStart());
+            castToBool(loadedInstr, fromType, exprType, expr->getLocStart());
```

Comment:
What happens if `exprType` is vector or matrix with single element i.e., `bool1` or `bool1x1`? Is this code fine for those cases as well?

Reply:
I should work properly. `castToBool` does support vec1 and mat1x1.